src/plots/cartesian/axes.js

```diff
@@ -3957,7 +3970,7 @@ function anyCounterAxLineAtZero(gd, ax, counterAxis, rng) {
             return typeof pos2 === 'number' && Math.abs(pos2 - zeroPosition) < tolerance;
         }
 
-        if(closeEnough(ax2._mainLinePosition) || closeEnough(ax2._mainMirrorPosition)) {
+        if(closeEnough(ax2._mainLinePositionShift) || closeEnough(ax2._mainMirrorPosition)) {
```

Comment:
Let's revert this part concerning zerolines:
```suggestion
        if(closeEnough(ax2._mainLinePosition) || closeEnough(ax2._mainMirrorPosition)) {
```

Reply:
Done! That did the trick